### ISSUE TITLE
DBeaver - java.lang.ClassNotFoundException: org.apache.commons.pool2.PooledObjectFactory

### ISSUE THREAD

**Issue by one70six** [diennea/herddb#512]
**BUG REPORT**
Attempting to use a Generic JDBC connection from the DBeaver Version 6.2.5 query tool to HerdDB v0.13.0 results in the following error:
```java.lang.ClassNotFoundException: org.apache.commons.pool2.PooledObjectFactory```

- What did you do?
  1. I followed the HerdDB documentation for [Getting Started](https://github.com/diennea/herddb/wiki/Getting-started) and [Using JDBC Driver](https://github.com/diennea/herddb/wiki/Using-JDBC-Driver)
  2. I used those instructions along with the DBeaver instructions [Database Drivers](https://github.com/dbeaver/dbeaver/wiki/Database-drivers) to attempt a connection
    - JDBC URL: jdbc:herddb:server:localhost:7000
    - Class Name: herddb.jdbc.Driver
    - User name: sa
    - Password: hdb
  3. Attempts were made using the ```herddb-jdbc-0.12.2.jar``` and ```herddb-jdbc-0.13.0.jar``` drivers by downloading the zipped release, unpacking, and targeting each
---
- What did you expect to see?
  1. Connection Successful from DBeaver
  2. **Both drivers work via Java code directly if you add the Maven dependency.**
---
- What did you see instead?
  1. In the ```dbeaver-debug.log``` file:
```bash
2019-11-26 09:39:47.882 - Connect with 'jdbc:herddb:server:localhost:7000' (A610477A-F3A8-0CFF-5201-CD1E9D9D6F1C-16ea85e6115-7ffb77075dcee2ad)
2019-11-26 09:39:47.887 - Connection failed (A610477A-F3A8-0CFF-5201-CD1E9D9D6F1C-16ea85e6115-7ffb77075dcee2ad)
2019-11-26 09:39:47.891 - org.jkiss.dbeaver.model.exec.DBCConnectException: Unexpected driver error occurred while connecting to the database
org.jkiss.dbeaver.model.exec.DBCConnectException: Unexpected driver error occurred while connecting to the database
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.openConnection(JDBCDataSource.java:191)
	at org.jkiss.dbeaver.ext.generic.model.GenericDataSource.openConnection(GenericDataSource.java:124)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCExecutionContext.connect(JDBCExecutionContext.java:88)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCRemoteInstance.initializeMainContext(JDBCRemoteInstance.java:88)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCRemoteInstance.<init>(JDBCRemoteInstance.java:54)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.initializeRemoteInstance(JDBCDataSource.java:111)
	at org.jkiss.dbeaver.ext.generic.model.GenericDataSource.<init>(GenericDataSource.java:106)
	at org.jkiss.dbeaver.ext.generic.model.meta.GenericMetaModel.createDataSourceImpl(GenericMetaModel.java:69)
	at org.jkiss.dbeaver.ext.generic.GenericDataSourceProvider.openDataSource(GenericDataSourceProvider.java:95)
	at org.jkiss.dbeaver.registry.DataSourceDescriptor.connect(DataSourceDescriptor.java:836)
	at org.jkiss.dbeaver.runtime.jobs.ConnectJob.run(ConnectJob.java:70)
	at org.jkiss.dbeaver.ui.dialogs.connection.ConnectionWizard$ConnectionTester.run(ConnectionWizard.java:246)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:103)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.lang.NoClassDefFoundError: org/apache/commons/pool2/PooledObjectFactory
	at herddb.jdbc.Driver.ensureDatasource(Driver.java:113)
	at herddb.jdbc.Driver.connect(Driver.java:69)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.lambda$0(JDBCDataSource.java:154)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.openConnection(JDBCDataSource.java:163)
	... 13 more
Caused by: java.lang.ClassNotFoundException: org.apache.commons.pool2.PooledObjectFactory
	at java.base/java.net.URLClassLoader.findClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	... 17 more
```

**Comment by hamadodene** [comment]
I'm able to reproduce using the "thin" HerdDB jdbc jar

@one70six have you tried using the "fat" jdbc driver available in the release zip (inside the root of the zip)? Please note that the "fat" and "thin" jar files have the same name, but one is much bigger than the other (that "fat" one is in the order of 45Mb).

**Comment by one70six** [comment]
@hamadodene yes I used the FAT JAR to perform this test, ~48MB on my machine, and it came up with the same results. Thanks!

**Comment by eolivelli** [comment]
Can you try with this JAR ?
https://repo1.maven.org/maven2/org/herddb/herddb-jdbc/0.13.0/herddb-jdbc-0.13.0-uber.jar

I got it from Maven central, it should be the same as the one inside the zip file (it's ~48MB ).
It surely contains the .class file that is not found from the JVM.

If it does not work maybe it is a problem with DBeaver, some kind of clash on the classpath.

I haven't tried to reproduce it yet, I will do as soon as I can.

**Comment by one70six** [comment]
I figured it out, it was a misunderstanding on my end... I was looking for the JDBC JAR under:
```~\herddb-services-0.13.0\lib\herddb-jdbc-0.13.0.jar``` 

which is 48KB when I should have been targeting the FAT JAR mentioned under:
```~\herddb-services-0.13.0\herddb-jdbc-0.13.0.jar```
which is ~45MB, much better. Maybe I misread that in the documentation, who knows.

The uber JAR you provided works as well.

It works as expected now so thank you very much for the assist! I'm interested in adding HerdDB on top of my existing Apache Pulsar environment since it utilizes 2/3 of the architecture. Very impressed so far guys!

**Comment by eolivelli** [comment]
Cool.
Feel free to ask any questions and report issues.


Please close this issue if your problem is solved.

**Comment by eolivelli** [comment]
@one70six I have played more with DBeaver and HerdDB,
I have found a bunch of minor problems related to JDBC Metadata and I have prepared a fix
#516

We are going to cut a release soon (this week).
If you have more feedback from DBBeaver usage I will be happy to follow up.

If you see errors in Metadata handling maybe they are fixed with #516 

I didn't know about DBeaver before, we can prepare official support for it and have HerdDB listed by default.
